memory/src/address.rs

```diff
@@ -48,6 +48,26 @@ impl<T, U> Address<T, U> {
     }
 }
 
+impl<T, U> Address<T, U>
+where
+    Offset<usize, ()>: Into<Offset<T, ()>>,
+    T: Rem<T, Output = T>,
+    T: Zero,
+    T: PartialEq,
+{
+    /// Try casting an existing `Address` into an `Address` of a different type
+    ///
+    /// Succeeds only, if they have the same alignment
+    #[inline]
+    pub fn try_cast<V>(self) -> Result<Address<T, V>, ()> {
+        let align: T = Offset::from_items(align_of::<V>()).into().items();
+        if self.0 % align != T::ZERO {
+            return Err(());
+        }
+        Ok(Address(self.0, PhantomData))
```

Comment:
Could this function be accomplished with `memory::Address::{lower, raise}`? I think this is only used with the BootInfo address.

Reply:
It could, but I thought, it might be nice to have it in the lib for others to use also.
Also with `lower` it complicates things, because we can't use rusts type interference.

Reply:
If you don't have any strong arguments against it, I really would like to leave it as it is.